config.h

```diff
@@ -8,19 +8,25 @@
 #define DEFAULT_MAIL_COMMAND "/usr/bin/mailx"
 #define COMPRESS_COMMAND "/usr/contrib/bin/gzip"
 #define UNCOMPRESS_COMMAND "/usr/contrib/bin/gunzip"
-#define STATEFILE "/var/run/logrotate.status"
+#define OLDSTATEFILE "/var/run/logrotate.status"
+#define STATEDIR "/var/run/logrotate/"
+#define STATEFILE STATEDIR "logrotate.status"
```

Comment:
As I understand it, @glensc wants the state file to be configurable at build-time.  Should we also configure the old state file, or just use the current preprocessor logic to guess it?